packages/abi-coder/src/types/IGetCoder.ts

```diff
@@ -0,0 +1,19 @@
+import type { ResolvedAbiType } from '../ResolvedAbiType';
+import type { Coder } from '../encoding/coders/AbstractCoder';
+
+import type { TEncodingOptions } from './TEncodingOptions';
+
+/**
+ * A function that can be used to obtain spec adhering coders.
+ *
+ * @param resolvedAbiType - the resolved type to return a coder for.
+ * @param options - options to be utilized during the encoding process.
+ */
+export type TGetCoderFn = (resolvedAbiType: ResolvedAbiType, options?: TEncodingOptions) => Coder;
+
+/**
+ * An interface that contains a function that obtains coders
+ */
+export interface IGetCoder {
```

Comment:
You can remove this interface and change its users to this:
```ts
export function getCoders(
  components: readonly ResolvedAbiType[],
  options: TEncodingOptions,
  getCoder: TGetCoderFn
) {
  return components.reduce((obj, component) => {
    const o: Record<string, Coder> = obj;

    o[component.name] = getCoder(component, options);
    return o;
  }, {});
}

export function getEncodingStrategy(encoding: string = ENCODING_V0): TGetCoderFn {
  switch (encoding) {
    case ENCODING_V0:
      return getCoderV0;
    case ENCODING_V1:
      return getCoderV1;
    default:
      throw new FuelError(
        ErrorCode.UNSUPPORTED_ENCODING_VERSION,
        `Encoding version ${encoding} is unsupported.`
      );
  }
}

export abstract class AbiCoder {
  static getCoder(
    abi: JsonAbi,
    argument: JsonAbiArgument,
    options: TEncodingOptions = {
      isSmallBytes: false,
    }
  ): Coder {
    const resolvedAbiType = new ResolvedAbiType(abi, argument);
    return getEncodingStrategy(options.encoding)(resolvedAbiType, options);
  }
...
}
```

Reply:
This is nice 👍🏻

Reply:
I'll admit I did like the explicitness of calling `getCoder` rather than the return value of `getEncodingStrategy`, but I agree with your sentiment. May just reword the functions.